LayoutEditor:Add menu to change toolbar coordinates format

geowar1 opened the pull request:
Contextual-(right- or control-)click on the coordinates in the toolbar to popup a menu that allows you to change between pixel (default), metric (cm) or English (feet, inches) formats.
Note: Current also changes grid intervals (metric: 10/10, english 12/16)… but doesn't have to.
TODO: add code to save/load format preferences.

bobjacobsen commented on java/src/jmri/jmrit/display/layoutEditor/LayoutEditorToolBarPanel.java:
```diff
@@ -359,6 +362,57 @@ protected void setupComponents() {
+            @Override
+            public void mousePressed(MouseEvent me) {
+                if (me.isPopupTrigger()) {
+                    locationPopupMenu.show(locationPanel, me.getX(), me.getY());
```
Please check for the isPopupTrigger in all three of `mousePressed(MouseEvent e)`, `mouseReleased(MouseEvent e)` and `mouseClicked(MouseEvent e)`.  Different look-and-feels use the mouse differently, so by checking all three you make sure that the popup gesture will be recognize across all types.

geowar1 replied:
Done! Good to know! Thanks!

dsand47 reviewed (commented):
Does there need to be some visual indication that there is clickable text on the tool bar?

geowar1 commented:
> Does there need to be some visual indication that there is clickable text on the tool bar?

… Perhaps… what do you suggest?

silverailscolo commented:
I would expect to see the units as currently set, either as part of the entry (eg. "12px", number formatting) or to the right of the amount field(s) so it's readable together with the numbers. In the latter case, place it on a button, or underline on a label to keep it plain?
Both ways will provide some clue for action.
Extra: if unit is shown as part of the entry, could user type an amount and a different unit and see it converted to the set unit on Enter/Tab?

geowar1 commented:
Ok, I moved the toolbar location coordinates format menu items from the popup to an "Options" sub-menu. 
Also added the code to store/restore them as per-(LayoutEditor)window preferences.
If/when this passes CI (and peer review) it's ready to merge. ;-)

dsand47 reviewed (approved):
Adding a menu item works nicely.
